examples/gno.land/p/demo/audio/biquad/biquad.gno

```diff
@@ -0,0 +1,57 @@
+package biquad
+
+import "math"
+
+type Filter struct {
+	a1, a2, b0, b1, b2     float64
+	x1_f, x2_f, y1_f, y2_f float64
+}
+
+func New(fc float64, fs float64, q float64, db float64, filter_type string) (f *Filter) {
+	w0 := 2 * 3.1415926535897932384626 * (fc / fs)
+	cosW := math.Cos(w0)
+	sinW := math.Sin(w0)
+	A := 1.0 // db = 0
+	_ = A
+	// A := math.Pow(10, (db / 40))
+	alpha := sinW / (2 * q)
+	beta := 1 // db = 0
+	// beta := A ^ 0.5/q
+	_ = beta
+	b0 := (1 - cosW) / 2
+	b1 := 1 - cosW
+	b2 := (1 - cosW) / 2
+	a0 := 1 + alpha
+	a1 := -2 * cosW
+	a2 := 1 - alpha
+	if filter_type == "highpass" {
+		b0 = (1 + cosW) / 2
+		b1 = -(1 + cosW)
+		b2 = (1 + cosW) / 2
+	} else if filter_type == "lowpass" {
+		// do nothing, default
+	}
```

Comment:
black magic; can it be demystified? or at least have a reference to the implemented algorithm?

Reply:
Ha yeah! It does indeed look like black magic. Its basically ways to derive coefficients to the solution of a "biquadratic" equation, which allows you to extract multiple filter types from the same topology, just changing coefficients. I added a reference that explains it well.